src/SeleniumLibrary/keywords/screenshot.py

```diff
@@ -201,16 +215,31 @@ def _get_screenshot_path(self, filename):
             directory = self._screenshot_root_directory or self.log_dir
         else:
             directory = self.log_dir
-        filename = filename.replace("/", os.sep)
+        filename = filename.replace("/", os.sep) 
+
         index = 0
         while True:
-            index += 1
-            formatted = _format_path(filename, index)
-            path = os.path.join(directory, formatted)
-            # filename didn't contain {index} or unique path was found
-            if formatted == filename or not os.path.exists(path):
-                return path
-
+            if  RAND in filename:
+                indexation = str(random.randint(LOW_LIMIT_RANDOM, UPPER_LIMIT_RANDOM))
+                formatted = _format_pathr(filename, indexation)
+                path = os.path.join(directory, formatted)
+                if formatted == filename or not os.path.exists(path):
+                    return path
+            elif  TIMESTAMP in filename:
+                indexation = datetime.now().strftime("%Y-%m-%d-%H-%M-%S")
+                formatted = _format_patht(filename, indexation)
+                path = os.path.join(directory, formatted)
+                # filename didn't contain {index} or unique path was found
+                if formatted == filename or not os.path.exists(path):
+                    return path       
+            elif 'index' in filename:
+                index+=1
+                formatted = _format_path(filename, index)
+                path = os.path.join(directory, formatted)
+                # filename didn't contain {index} or unique path was found
+                if formatted == filename or not os.path.exists(path):
+                    return path       
+            
```

Comment:
Just a quick read of this I believe there is a logic error here where if none of the three are in the filename one will get themselves into an infinite loop.